fabric-api-base/src/main/java/net/fabricmc/fabric/api/event/Event.java

```diff
@@ -88,4 +88,15 @@ public void register(Identifier phase, T listener) {
 	public void addPhaseOrdering(Identifier firstPhase, Identifier secondPhase) {
 		// This is not abstract to avoid breaking existing Event subclasses, but they should really not be subclassing Event.
 	}
+
+	/**
+	 * Returns whether the event has any listeners. Useful for skipping steps related to
+	 * event invocation, such as making context objects, when the event has no listeners.
+	 * This method is unnecessary when the only code guarded by this check is the event invocation.
+	 *
+	 * @return whether the event has any listeners
+	 */
+	public boolean hasListener() {
```

Comment:
IMO this is a bad pattern. If you are concerned about allocation, reuse the context object. In typical modpacks, most events will be used and these checks will be pointless anyway.

Reply:
Thats a fair point, @apple502j did you have a specific use case in mind?

Reply:
A usecase I had for this was for a deprecated event, where actually properly servicing it required setting up some expensive stuff

Reply:
Also, not all context objects can be reused.

Reply:
Can you link some specific examples?

Reply:
https://github.com/BasiqueEvangelist/ne-v-seti/blob/sanity-1.17/src/main/java/me/basiqueevangelist/nevseti/OfflineDataCache.java#L30
I guess I misremembered specifically how much it would take to service this event, but I think it will still be useful for events that are deprecated and require running extra logic to run their invokers